Added typescript usage example. (#11)
jeffijoe/mobx-task: README.md

Alternatively, use `this.boo = this.boo.bind(this)` in the constructor.

## Using with `typescript`

> Best way to work with typescript is to install `@types/mobx-task`. Definitions covers most use cases. The tricky part are

Reviewer: Weird newlines, and "the tricky part is" 😄 

```suggestion
Best way to work with typescript is to install `@types/mobx-task`. Definitions covers most use cases. The tricky part is
```

> decorators because they are not able to change type of decorated element. You will have to do type assertion or use plain

Reviewer: ```suggestion
decorators because they are not able to change the type of the decorated target. You will have to do type assertion or use plain
```

observables.

```
npm install --save-dev @types/mobx-task
```

Example:

```ts
class Test {
  @task taskClassMethod(arg1: string, arg2: number) {
    let result: boolean
    ...
    return result
  }
  
  @task assertTypeHere = <Task<boolean, [string, number]>>(arg1: string, arg2: number) => {
    let result: boolean
    ...
    return result
  }
  
  @task assertTypeHereWithAs = (arg1: string, arg2: number) => {
    let result: boolean
    ...
    return result
  } as Task<boolean, [string, number]>
}

const test = new Test()

// dont care about task methods, props and return value and type
const doSomething = async () => {
  await test.taskClassMethod('a', 1)
  ...
}

// want to use task props and returned promise

> (test.taskClassMethod as Task).then(...) // Task<any, any[]>

Reviewer: I believe this change is needed to make it semantically correct?

```suggestion
(test.taskClassMethod as Task)("one", 2).then(...) // Task<any, any[]>
```
